codegen-server/typescript/src/main/kotlin/software/amazon/smithy/rust/codegen/server/typescript/smithy/RustServerCodegenTsPlugin.kt

```diff
@@ -0,0 +1,90 @@
+        // - writer: The active RustWriter at the given location
+        val codegenDecorator: CombinedServerCodegenDecorator =
+            CombinedServerCodegenDecorator.fromClasspath(
+                context,
+                CombinedServerCodegenDecorator(DECORATORS + ServerRequiredCustomizations()),
+            )
+
+        // TsServerCodegenVisitor is the main driver of code generation that traverses the model and generates code
+        logger.info("Loaded plugin to generate Rust/Node bindings for the server SSDK for projection ${context.projectionName}")
+        TsServerCodegenVisitor(context, codegenDecorator).execute()
+    }
+
+    companion object {
+        /**
+         * When generating code, smithy types need to be converted into Rust types—that is the core role of the symbol provider
+         *
+         * The Symbol provider is composed of a base [SymbolVisitor] which handles the core functionality, then is layered
+         * with other symbol providers, documented inline, to handle the full scope of Smithy types.
+         */
+        fun baseSymbolProvider(
+            model: Model,
+            serviceShape: ServiceShape,
+            symbolVisitorConfig: SymbolVisitorConfig,
+            constrainedTypes: Boolean = true,
+        ) =
+            TsServerSymbolVisitor(model, serviceShape = serviceShape, config = symbolVisitorConfig)
+                // Generate public constrained types for directly constrained shapes.
+                // In the Typescript server project, this is only done to generate constrained types for simple shapes (e.g.
+                // a `string` shape with the `length` trait), but these always remain `pub(crate)`.
+                .let { if (constrainedTypes) ConstrainedShapeSymbolProvider(it, model, serviceShape) else it }
+                // Generate different types for EventStream shapes (e.g. transcribe streaming)
+                .let { EventStreamSymbolProvider(symbolVisitorConfig.runtimeConfig, it, model, CodegenTarget.SERVER) }
+                // Add Rust attributes (like `#[derive(PartialEq)]`) to generated shapes
+                .let { BaseSymbolMetadataProvider(it, model, additionalAttributes = listOf()) }
+                // Constrained shapes generate newtypes that need the same derives we place on types generated from aggregate shapes.
+                .let { ConstrainedShapeSymbolMetadataProvider(it, model, constrainedTypes) }
+                // Streaming shapes need different derives (e.g. they cannot derive Eq)
+                .let { TsStreamingShapeMetadataProvider(it, model) }
+                // Derive `Eq` and `Hash` if possible.
+                .let { DeriveEqAndHashSymbolMetadataProvider(it, model) }
+                // Rename shapes that clash with Rust reserved words & and other SDK specific features e.g. `send()` cannot
+                // be the name of an operation input
+                .let { RustReservedWordSymbolProvider(it, model) }
+    }
+}
```

Comment:
@pose please take a look to how this is done in Python now as the API changed

Reply:
Done. Seems like `model` was removed from almost everywhere.